### PR TITLE
Added linux and macos support (+windows fixes)

### PR DESCRIPTION
**Pull request by gbuzogany**
Let me know if there's anything broken

### CODE REVIEW

**Comment by zaucy** [comment]
Thanks for your contribution! This looks great!
I'll take a look at it later today and get it merged in.

File: BUILD.bazel [hunk 1]
```diff
@@ -54,9 +66,45 @@ cc_library(
...
+		":SDL2_lib",
+	],
+	linkopts = SDL_LINKOPTS,
+	visibility = ["//visibility:public"],
+)
+
 cc_library(
-	name = "SDL2",
-	hdrs = glob(["include/**/*.h"]),
+	name = "SDL2_lib",
```
Comment (zaucy) [inline BUILD.bazel:105]: Is it possible to leave this target name as "SDL2"? A bit of a nicer name for anyone who is using SDL2 statically.
```suggestion
	name = "SDL2",
```

File: BUILD.bazel [hunk 2]
```diff
@@ -54,9 +66,45 @@ cc_library(
...
+	platform_type = "macos",
+	visibility = ["//:__subpackages__"],
+)
+
+genrule(
+   name = "SDL2_dylib",
+   srcs = ["SDL2_mac"],
+   outs = ["SDL2.dylib"],
+   cmd = "cp $(location SDL2_mac) $(location SDL2.dylib)",
+)
```
Comment (zaucy) [inline BUILD.bazel]: A bit of inconsistent indenting here

File: BUILD.bazel [hunk 3]
```diff
@@ -20,12 +30,14 @@ SDL_LINKOPTS = select({
...
 		"-DEFAULTLIB:setupapi",
 		"-DEFAULTLIB:opengl32",
+		"-DEFAULTLIB:kernel32",
 	],
 	"//conditions:default": [],
 })
 
 cc_library(
 	name = "SDL2_hdrs",
+	include_prefix = "SDL2",
```
Comment (zaucy) [inline BUILD.bazel]: Isn't the inclusion of SDL2 typically done with

```cpp
#include "SDL.h"
// or
#include <SDL.h>
```

Not

```cpp
#include "SDL2/SDL.h"
// or
#include <SDL2/SDL.h>
```

```suggestion
```

File: LICENSE.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+including commercial applications, and to alter it and redistribute it
+freely, subject to the following restrictions:
+
+1. The origin of this software must not be misrepresented; you must not
+   claim that you wrote the original software. If you use this software
+   in a product, an acknowledgment in the product documentation would be
+   appreciated but is not required.
+2. Altered source versions must be plainly marked as such, and must not be
+   misrepresented as being the original software.
+3. This notice may not be removed or altered from any source distribution.
```
Comment (zaucy) [inline LICENSE.txt]: The license is already dicated by the `COPYING.txt` file from the original SDL2 source code. Is this necessary?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+The build succeeds, but I didn't try anything more than that.
+
+# Windows build:
+
+```
+bazel build //:SDL2.dll
+```
+
+The build succeeds, but I didn't try anything more than that.
```
Comment (zaucy) [inline README.md]: Could we remove this file from the the pull request? I want to keep this repository as similar to the original SDL2 source code as possible.

If we were to add a README I think I would like it to include installation instructions, but even then I'm unsure.

**Review by zaucy — CHANGES_REQUESTED** [review]
Thanks for your hard work! Just some minor changes!

I wans't able to build `//:SDL2.dll` on my windows machine. There are a bunch of unresolved externals. Were you able to get it to build? Is there some extra step that I am msising? I should get around to setting up some CI so these things can be verified automatically.

File: BUILD.bazel (hunk 1 above)
Comment (gbuzogany) [reply BUILD.bazel:105]: If it's named "SDL2", then the windows build of "SDL2.dll" fails (naming conflict on bazel).
One of the two has to be renamed (or a genrule has to be created, or something else), and I just chose the static one, but I'm open to suggestions.

File: BUILD.bazel (hunk 2 above)
Comment (gbuzogany) [reply BUILD.bazel]: Well spotted!

File: BUILD.bazel (hunk 3 above)
Comment (gbuzogany) [reply BUILD.bazel]: Yep, I've been using it wrong for a long time. Well spotted again!

File: LICENSE.txt (hunk 4 above)
Comment (gbuzogany) [reply LICENSE.txt]: Was just following common open-source repo structure, but I'll get rid of it.

File: README.md (hunk 5 above)
Comment (gbuzogany) [reply README.md]: Moved it to README-BAZEL.md, but I can remove it if you want.
As this repo is WIP (and barely tested by me at least), I believe it's important to align people's expectations to what to find here, and avoid wasting time of a random stranger that can't make it work.
It's a good home to describe usage, limitations, etc, and I think the risk of collision with the main SDL2 is minimal.

**Comment by gbuzogany** [comment]
Indeed, having it on a CI would be wonderful. Apparently some changes I made for linux broke the Windows build, but now I believe it's sorted out.

File: README.md (hunk 5 above)
Comment (zaucy) [reply README.md]: You're right, having a README here would be nice for that. Feel free to keep it as README.md. I may make some change to it after this merge, however. :)

File: BUILD.bazel (hunk 1 above)
Comment (zaucy) [reply BUILD.bazel:105]: Ah that's unfortunate. With that constraint `SDL2_lib` is a decent name then!

**Comment by zaucy** [comment]
I'll get around to testing this later today! Thanks again!

**Comment by gbuzogany** [comment]
I used it with a windows project, and it worked! :)
I also found problems with it on linux :(
I'll let you know when I get it going on linux

**Comment by gbuzogany** [comment]
Fixed some issues on linux. 
Now I'm able to link it (static/dynamic) to a project without any issues, but it's not behaving properly (not managing to initialize video).

So, for now:
- Works (at least partially) on windows.
- Works (at least partially) on macos.
- Compiles, links, but doesn't initalize video properly on linux.

I think it's a good time to start running the SDL tests to track down issues.
I'd say you can go on with the merge, and we go for the tests on another issue.

**Comment by zaucy** [comment]
I wasn't able to build on linux with these changes, but I was able to build on windows. (I don't have a mac to test macos so I'll trust it's working ;P)

There are some great improvements so I'll merge this open, but leave the issue #7 open until we can confirm it's building on windows, linux and macOS.

**Pull request merged**